src/ConstructiveSolidGeometry/VolumePrimitives/Torus.jl

```diff
@@ -66,14 +66,86 @@ This is a `Torus` with `r_tube` having an inner radius of 1 and an outer radius
 
 See also [Constructive Solid Geometry (CSG)](@ref).
 """
-@with_kw struct Torus{T,CO,TR,TP<:Union{Nothing,T},TT,TT1,TT2} <: AbstractVolumePrimitive{T,CO}
-    r_torus::T = 1
-    r_tube::TR = 1  # (r_tube_in, r_tube_out)
-    φ::TP = nothing
-    θ::TT = nothing
-
-    origin::CartesianPoint{T} = zero(CartesianPoint{T})
-    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+struct Torus{T,CO,TR,TP<:Union{Nothing,T},TT,TT1,TT2} <: AbstractVolumePrimitive{T,CO}
+    r_torus::T
+    r_tube::TR  # (r_tube_in, r_tube_out)
+    φ::TP 
+    θ::TT
+
+    origin::CartesianPoint{T}
+    rotation::SMatrix{3,3,T,9}
+end
+
+#Type conversion happens here
+function Torus{T,CO}(r_torus, r_tube, φ::Tuple{T,T}, θ, origin, rotation) where {T,CO}
+    r_torus = _csg_convert_args(T, r_torus)
+    r_tube = _csg_convert_args(T, r_tube)
+    φ = _csg_convert_args(T, φ)
+    _φ = abs(φ[2]-φ[1])
+    rotation=rotation*SMatrix{3}(cos(φ[1]),sin(φ[1]),0,-sin(φ[1]),cos(φ[1]),0,0,0,1)
+    θ = _csg_convert_args(T, θ)
+    TT1, TT2 = _get_conemantle_type(θ)
+    Torus{T,CO,typeof(r_tube),typeof(_φ),typeof(θ),TT1,TT2}(r_torus, r_tube, _φ, θ, origin, rotation)
+end
+
+function Torus{T,CO}(r_torus, r_tube, φ, θ, origin, rotation) where {T,CO}
+    r_torus = _csg_convert_args(T, r_torus)
+    r_tube = _csg_convert_args(T, r_tube)
+    φ = _csg_convert_args(T, φ)
+    θ = _csg_convert_args(T, θ)
+    TT1, TT2 = _get_conemantle_type(θ)
+    Torus{T,CO,typeof(r_tube),typeof(φ),typeof(θ),TT1,TT2}(r_torus, r_tube, φ, θ, origin, rotation)
+end
```

Comment:
We only want one method here. That's a lot of code duplication.

We would need to add some general function to process and handle `φ` and `rotation` inside it.
We also need it for other primitives I believe.